youtube_dl/extractor/beampro.py

```diff
@@ -0,0 +1,89 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    int_or_none,
+    ExtractorError,
+    compat_str,
+    clean_html,
+    parse_iso8601,
+)
+
+
+class BeamProLiveIE(InfoExtractor):
+    IE_NAME = 'Beam:live'
+    _VALID_URL = r'https?://(?:\w+.)?beam.pro/(?P<id>[^?]+)$'
+    _API_CHANNEL = 'https://beam.pro/api/v1/channels/{0}'
+    _API_MANIFEST = 'https://beam.pro/api/v1/channels/{0}/manifest.{1}'
+    _VALID_MANIFESTS = ('smil', 'm3u8', 'light', 'light2', 'ftl', 'ftlOld')
```

Comment:
I'm planning to expand this extractor later. There are lots of stuff to do: VODs, various playlists like channels, categories, etc., etc.